core/src/components/refresher/test/basic/e2e.ts

```diff
@@ -1,10 +1,28 @@
+import type { E2EPage } from '@stencil/core/testing';
 import { newE2EPage } from '@stencil/core/testing';
 
-test('refresher: basic', async () => {
-  const page = await newE2EPage({
-    url: '/src/components/refresher/test/basic?ionic:_testing=true'
+import { pullToRefresh } from '../test.utils';
+
+describe('refresher: basic', () => {
+
+  let page: E2EPage;
+
+  beforeEach(async () => {
+    page = await newE2EPage({
+      url: '/src/components/refresher/test/basic?ionic:_testing=true'
+    });
+  });
+
+  it('should match existing visual screenshots', async () => {
+    const compare = await page.compareScreenshot();
+    expect(compare).toMatchScreenshot();
+  });
+
+  it('should load more items when performing a pull-to-refresh', async () => {
+    await pullToRefresh(page);
```

Comment:
This is a bit tricky because there are 3 different PTR types: legacy, iOS native, and MD native. Which one is this designed to test?

Reply:
This should test iOS and MD, unless they are somehow configured through a property instead of the `mode`? How do you render the refresher in the legacy type?

This test emulates a mouse click/drag down from the center of the screen, which should fire the refresher events for each mode the same.

Reply:
Customizing the pullingIcon should trigger legacy refreshers. The iOS native refresher only works on a device with rubber band scrolling, so we probably do not have to worry about it.

As long as we have one test where we customize the pulling icon and one where we do not, that should be fine.